Tests/StencilTests/EnvironmentSpec.swift

```diff
@@ -1,342 +1,392 @@
-import XCTest
-import Spectre
 import PathKit
+import Spectre
 @testable import Stencil
+import XCTest
 
-class EnvironmentTests: XCTestCase {
-  func testEnvironment() {
-    describe("Environment") {
-      var environment: Environment!
-      var template: Template!
+final class EnvironmentTests: XCTestCase {
+  var environment = Environment(loader: ExampleLoader())
+  var template: Template = ""
 
-      $0.before {
-        environment =  Environment(loader: ExampleLoader())
-        template = nil
-      }
+  override func setUp() {
+    super.setUp()
 
-      $0.it("can load a template from a name") {
-        let template = try environment.loadTemplate(name: "example.html")
-        try expect(template.name) == "example.html"
-      }
+    let errorExtension = Extension()
+    errorExtension.registerFilter("throw") { (_: Any?) in
+      throw TemplateSyntaxError("filter error")
+    }
+    errorExtension.registerSimpleTag("simpletag") { _ in
+      throw TemplateSyntaxError("simpletag error")
+    }
+    errorExtension.registerTag("customtag") { _, token in
+      ErrorNode(token: token)
+    }
 
-      $0.it("can load a template from a names") {
-        let template = try environment.loadTemplate(names: ["first.html", "example.html"])
-        try expect(template.name) == "example.html"
-      }
+    environment = Environment(loader: ExampleLoader())
+    environment.extensions += [errorExtension]
+    template = ""
+  }
 
-      $0.it("can render a template from a string") {
-        let result = try environment.renderTemplate(string: "Hello World")
-        try expect(result) == "Hello World"
-      }
+  func testLoadingRendering() {
```

Comment:
testLoading**And**Rendering, though I wonder if we better split it into two tests.

Reply:
👍